src/main/java/org/sonatype/repository/conan/internal/metadata/ConanAbsoluteUrlIndexer.java

```diff
@@ -0,0 +1,208 @@
+ * Eclipse Foundation. All other trademarks are the property of their respective owners.
+ */
+package org.sonatype.repository.conan.internal.metadata;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.annotation.Nullable;
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import org.sonatype.goodies.common.ComponentSupport;
+import org.sonatype.nexus.repository.Repository;
+import org.sonatype.nexus.repository.storage.Asset;
+import org.sonatype.nexus.repository.storage.Bucket;
+import org.sonatype.nexus.repository.storage.StorageFacet;
+import org.sonatype.nexus.repository.storage.StorageTx;
+import org.sonatype.nexus.repository.storage.TempBlob;
+import org.sonatype.nexus.repository.transaction.TransactionalStoreMetadata;
+import org.sonatype.nexus.repository.view.Content;
+import org.sonatype.nexus.repository.view.payloads.StringPayload;
+import org.sonatype.nexus.transaction.UnitOfWork;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.base.Supplier;
+
+import static java.nio.charset.Charset.defaultCharset;
+import static java.util.Collections.emptyMap;
+import static org.sonatype.nexus.repository.storage.AssetEntityAdapter.P_ASSET_KIND;
+import static org.sonatype.nexus.repository.view.Content.maintainLastModified;
+import static org.sonatype.nexus.repository.view.ContentTypes.APPLICATION_JSON;
+import static org.sonatype.repository.conan.internal.AssetKind.CONAN_INDEX;
+import static org.sonatype.repository.conan.internal.proxy.ConanProxyHelper.HASH_ALGORITHMS;
+import static org.sonatype.repository.conan.internal.proxy.ConanProxyHelper.findAsset;
+import static org.sonatype.repository.conan.internal.proxy.ConanProxyHelper.toContent;
+
+/**
+ * download_url files contain absolute paths to each asset
+ *
+ * This class removes the absolute address so as to redirect back to this repository
+ *
+ * @since 3.conan
+ */
+@Singleton
+@Named
+public class ConanAbsoluteUrlIndexer
+    extends ComponentSupport
+{
+  private static final ObjectMapper MAPPER = new ObjectMapper();
+
+  @Nullable
+  public TempBlob updateAbsoluteUrls(final TempBlob tempBlob,
+                                     final Repository repository,
+                                     final String assetName) {
+    Map<String, URL> downloadUrlContents = readIndex(tempBlob.get(), assetName);
+    Map<String, URL> indexes = new HashMap<>();
+
+    for (Map.Entry<String, URL> entry : downloadUrlContents.entrySet()) {
+      URL originalUrl = entry.getValue();
+      URL indexUrl = getIndexedUrl(repository.getUrl(), entry.getValue().getPath());
```

Comment:
The `repository.getUrl()` could theoretically change if the NXRM base url is changed so I don't know if this is really what you want to be doing if you're saving this content. You _do_ need to do this, but my point is that the rewritten url could end up being invalid and there's no way (no task, nothing) to rebuild it when that happens, infrequent occurrence though it may be.